clients/src/main/java/org/apache/kafka/common/requests/MetadataResponse.java

```diff
@@ -72,13 +76,24 @@
     private static final String ISR_KEY_NAME = "isr";
 
     private final Collection<Node> brokers;
+    private final Node controller;
     private final List<TopicMetadata> topicMetadata;
 
+    /**
+     * Constructor for the latest version
+     */
+    public MetadataResponse(List<Node> brokers, int controllerId, List<TopicMetadata> topicMetadata) {
```

Comment:
These constructors are generally only used in the broker. I like passing in the version because then one constructor can handle all versions and the broker is guaranteed to send the correct version back without a bunch of if/switch logic.

There could be a case in the future where a separate constructor is required to maintain compatibility. But if I can avoid many constructors I would prefer it.